### DISCUSSION TITLE
History of Ph, ORP, and Dosing?

### DISCUSSION THREAD

**Discussion by davthomaspilot** [tagyoureit/nodejs-poolController#745] [REM] [answered]
Is there a way to display ph and ORP versus time?  And, dosing volume/dosing time versus time?

I'm familiar with packet capture and display, is that what I should use?

Thanks!

**Comment by rstrouse** [comment]
I use InfluxDB and Grafana.

**Reply by davthomaspilot** [reply]
So, I think I need to parse the packets and write them to a database?  Then, write queries to get the data and find a package to plot?

I can do all that--I'd used SqlLite, since I used that many times.   But, just wanted to make sure I wasn't missing something built in.

Thanks!

**Reply by davthomaspilot** [reply]
Also, can I truncate --size=0 the file that has the packets after writing data to the db?  Or, might that cause issues with njsPC?

**Reply by johnny2678** [reply]
Hey @davthomaspilot - strongly recommend the Influx integration that's already built in to njspc.  Then you can slap grafana on top of that for visualizations.

**Reply by rstrouse** [reply]
Are you referring to the RS485 packets?  If you are then you shouldn't.  Those packets need context and only in the context of the moment will they make sense.  Why not InfluxDB?  This is a time series database that is efficient for this sort of thing and the interface is built into njsPC.  Grafana is a plotting package that loads it up.

**Reply by davthomaspilot** [reply]
Why not InfluxDB?  Only because of my ignorance.  Reading up on it now.

Found this:

https://www.troublefreepool.com/threads/any-pool-automation-grafana-influxdb-users-out-there.222946/

**Comment by davthomaspilot** [comment]
Ok, thanks.
Are there any resources on getting it set up?


    On Saturday, May 6, 2023 at 04:09:29 PM EDT, johnny2678 ***@***.***> wrote:  
 
 


Hey @davthomaspilot - strongly recommend the Influx integration that's already built in to njspc. Then you can slap grafana on top of that for visualizations.

**Comment by davthomaspilot** [comment]
I've installed influxdb and garfana.  

Followed a tutorial to create a database and add/test a data source in garfana.

What do I need to do in njsPC to get records written to a db?

**Comment by davthomaspilot** [comment]
Ok, I created a database pool and edited config.json for the influx userid and password and port 8086.  

Does this message in /var/log/syslog means records are being written?

May  7 08:31:34 PoolController influxd-systemd-start.sh[547]: [httpd] 192.168.0.31 - - [07/May/2023:08:31:34 -0400] "POST /api/v2/write?org=undefined&bucket=pool%2Fautogen&precision=ms HTTP/1.1 " 204 0 "-" "influxdb-client-js/1.33.2" 1ac210e2-ecd3-11ed-8010-b827eb8ab26b 13127

**Comment by davthomaspilot** [comment]
Must be working--I don't know grafana well enough to build a query yet, but I see field names that must have been populated from njsPC.  So, I think now need to learn grafana.

**Comment by johnny2678** [comment, marked as answer]
njsPC supports influx 1.x and 2.x.  I am old school and I absolutely refuse to update influx past 1.8.4 as 2.x is mostly a rewrite with completely new connection strings and query syntax.

Here's a quick scribe i did for setting up a 1.x connection.  Thought I would be able to embed in comments but apparently you have to pay for that so PDF it is...

[EnablingInfluxDBonnjsPC._PDF_2023-05-07124312.448250.pdf](https://github.com/tagyoureit/nodejs-poolController/files/11414995/EnablingInfluxDBonnjsPC._PDF_2023-05-07124312.448250.pdf)


edit: it does appear I can share [the url](https://scribehow.com/shared/Enabling_InfluxDB_on_njsPC__iL89vQKyTM2aopKXAsWdyg)

**Reply by davthomaspilot** [reply]
Thanks!

I'll devour it now.

So far, I can see tables got creating by issuing SQL like queries in the influx cli.  But, the tables are empty.

Still learning...

**Reply by davthomaspilot** [reply]
Well, I'd done all that.  

But I do see data in the tables now, using select * from the Influx CLI!

Not sure if this was the fix, but I saw a message in /var/log about not having LAT and LON set.  So, I configured those.  Now I see data in the tables.

Thanks!

**Comment by davthomaspilot** [comment]
I tried out the garfana dashboard @tagyoureit had on.  Pretty cool!




Thanks!

**Reply by johnny2678** [reply]
most of the ones I've seen live here: https://grafana.com/grafana/plugins/

Which ones are missing?

**Comment by davthomaspilot** [comment]
I found them.  

D3 Gauge by Brian Gahn and (I think) Discrete by Natal Energy.

Great way to get started--I just had to edit the panels to reference  InfluxDB-1 data source instead of InfluxDB--I guess I created one when I was doing a tutorial.

Then I deleted the things my pool doesn't have, like pressure, and added ORP tank and dosing by copy/pasting the ph equivalents then editing.

Everything I ever wanted!!

**Reply by johnny2678** [reply]
yea, shameful admission that part of my job is to sell enterprise analytics software and I find grafana way more intuitive 😂

**Reply by tagyoureit** [reply]
you sell enterprise analytics, too?  You must be my competition.  :)  I work for Tableau.

**Reply by johnny2678** [reply]
haha, then I guess a belated welcome to the team is in order??  😂😂  I work for Salesforce.

I'm actually on the Solution Engineering side, which makes me just technical enough to tackle projects like pool automation :)

**Reply by tagyoureit** [reply]
Look me up in Slack and shoot me a note!  https://github.com/tagyoureit/nodejs-poolController/blob/master/app.ts#L3

**Comment by ibehren1** [comment]
Working on my own Grafana dashboard.  Willing to share it if there is interest.
(Don't mind my out of balance numbers... just opened for the year and not in balance yet.)

![image](https://user-images.githubusercontent.com/7774746/236984312-02e875b7-0354-4f57-a88c-565e1b5ef571.png)

**Reply by davthomaspilot** [reply]
I'm interested.

**Reply by johnny2678** [reply]
nice!  I'll share mine that I've built up over the last 5-6 years.  

I have a few that I check regularly.  First, an operational monitoring dashboard.  This tells me what's going on with the pool now/today so far:
![screencapture-192-168-5-140-3000-d-BbYva0wGk-pool-dashboard-w-spa-2021-2023-05-09-12_28_22](https://github.com/tagyoureit/nodejs-poolController/assets/8219743/29a22445-d780-484b-83ad-edcf135ee230)


Then I have a Seasonal Dashboard I use to monitor Energy/Temps over time:
![screencapture-192-168-5-140-3000-d-000000003-pool-seasonal-dashboard-2023-05-09-12_34_53](https://github.com/tagyoureit/nodejs-poolController/assets/8219743/8a12579a-626d-4884-a313-7628703ef198)

Finally I have a chemistry dashboard I use to tweak my dosing:
![screencapture-192-168-5-140-3000-d-v4EoNPmiz-pool-chemistry-2023-05-09-12_29_46](https://github.com/tagyoureit/nodejs-poolController/assets/8219743/3a960a58-0e93-4193-b708-dec970a56391)

Normally, I'd be happy to share the panel def json but my njspc influx schema has some custom bindings/tags.  On top of that, I've been logging pool data to influx since before influx support was even included in njspc so this is a mixture of the njspc schema and my own schema and would be non-sensical to replicate at the db layer.  But posting anyway in the event it helps others.  Cheers!

**Reply by ChrisOgden** [reply]
What are the physical components used to measure each of these datapoints?
